spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -883,6 +917,17 @@ private void checkIdle() {
 			}
 		}
 
+		private void idleBetweenPollIfNecessary() throws InterruptedException {
```

Comment:
We still need to catch the interrupt and reset the interrupted flag because `handleConsumerException()` simply logs and eats the exception and it won't set the interrupted flag.

I meant that the choice was to just set the interrupted flag and exit (so something else deals with the interrupt), or throw the illegal state exception here and move the call to the try block.

Reply:
Pushed. Looks like I still need to learn how to handle `InterruptedException` properly 😄